lib/dynamoid/adapter_plugin/aws_sdk_v3.rb

```diff
@@ -562,10 +563,12 @@ def query_count(table_name, options = {})
       # @todo: Provide support for various options http://docs.aws.amazon.com/sdkforruby/api/Aws/DynamoDB/Client.html#scan-instance_method
       def scan(table_name, conditions = {}, options = {})
         return enum_for(:scan, table_name, conditions, options) unless block_given?
+
         table = describe_table(table_name)
 
         Scan.new(client, table, conditions, options).call.each do |page|
-          yield page.items.map{ |row| result_item_to_hash(row) }
+          yield page.items.map { |row| result_item_to_hash(row) },
+            { last_evaluated_key: page.last_evaluated_key }
```

Comment:
Am I right, now we can come back to ordinal enumerator like it was before? As far as we pass two arguments to `yield`?

Reply:
Good question, I'll give it a try.